src/System.Text.Json/tests/Serialization/SnakeCaseUnitTests.cs

```diff
@@ -0,0 +1,157 @@
+        [InlineData("i_phone", "IPhone")]
+        [InlineData("ip_hone", "IPHone")]
+        [InlineData("iph_one", "IPHOne")]
+        [InlineData("ipho_ne", "IPHONe")]
+        [InlineData("iphone", "IPHONE")]
+        //
+        [InlineData("id", "id")]
+        [InlineData("id", "ID")]
+        //
+        [InlineData("url", "url")]
+        [InlineData("url", "URL")]
+        [InlineData("url_value", "url_value")]
+        [InlineData("url_value", "URLValue")]
+        //
+        [InlineData("xml2json", "xml2json")]
+        [InlineData("xml2json", "Xml2Json")]
+        //
+        [InlineData("already_snake_case", "already_snake_case")]
+        [InlineData("_already_snake_case", "_already_snake_case")]
+        [InlineData("__already_snake_case", "__already_snake_case")]
+        [InlineData("already_snake_case_", "already_snake_case_")]
+        [InlineData("already_snake_case__", "already_snake_case__")]
+        //
+        [InlineData("sn_a__k_ec_as_e", "sn_a__k_ec_as_e")]
+        [InlineData("sn_ak_ec_as_e", "sn_ak_ec_as_e")]
+        [InlineData("sn_a__k_ec_as_e", "SnA__ kEcAsE")]
+        [InlineData("sn_a__k_ec_as_e", "SnA__kEcAsE")]
+        [InlineData("sn_ak_ec_as_e", "SnAkEcAsE")]
+        //
+        [InlineData("spaces", "spaces ")]
+        [InlineData("spaces", "spaces  ")]
+        [InlineData("spaces", "spaces   ")]
+        [InlineData("spaces", "   spaces")]
+        [InlineData("spaces", "  spaces")]
+        [InlineData("spaces", " spaces")]
+        //
+        [InlineData("9999_12_31t23_59_59_9999999z", "9999-12-31T23:59:59.9999999Z")]
+        [InlineData("hi_this_is_text_time_to_test", "Hi!! This is text. Time to test.")]
+        //
+        [InlineData("is_cia", "IsCIA")]
+        [InlineData("is_json_property", "IsJSONProperty")]
+        //
+        [InlineData("lower_case", "lower case")]
+        [InlineData("lower_case", "lower Case")]
+        [InlineData("lower_case", "lowerCase")]
+        [InlineData("lower_c_a_se", "lower cASe")]
+        [InlineData("lowe_r_case", "loweR case")]
+        [InlineData("lowe_r_case", "loweR Case")]
+        [InlineData("lowe_r_c_a_se", "loweR cASe")]
+        [InlineData("upper_case", "Upper case")]
+        [InlineData("upper_case", "Upper Case")]
+        [InlineData("upper_case", "UPPER CASE")]
+        [InlineData("upper_case", "UpperCase")]
+        [InlineData("upper_c_a_se", "Upper cASe")]
+        [InlineData("uppe_r_case", "UppeR case")]
+        [InlineData("uppe_r_case", "UppeR Case")]
+        [InlineData("uppe_r_c_a_se", "UppeR cASe")]
+        [InlineData("u_pper_case", "UPper case")]
+        [InlineData("u_pper_case", "UPper Case")]
+        [InlineData("u_pper_case", "UPperCase")]
+        [InlineData("u_pper_c_a_se", "UPper cASe")]
+        [InlineData("u_ppe_r_case", "UPpeR case")]
+        [InlineData("u_ppe_r_case", "UPpeR Case")]
+        [InlineData("u_ppe_r_c_a_se", "UPpeR cASe")]
+        //
+        [InlineData("ä", "ä")]
```

Comment:
Non-ascii tests like these should be written using escaped/hex characters.

> When including non-ASCII characters in the source code use Unicode escape sequences (\uXXXX) instead of literal characters. Literal non-ASCII characters occasionally get garbled by a tool or editor.

See: 
https://github.com/dotnet/corefx/blob/5cee7c97d602f294e27c582d4dab81ec388f1d7b/src/System.Text.Json/tests/Utf8JsonReaderTests.cs#L1993

So, for this case (same for the special "42" below):
```suggestion
        [InlineData("\u00E4", "\u00E4")]
```